modules/selection/main/index.coffee

```diff
@@ -271,7 +271,29 @@ class Selection
     this
 
   # clears the contents of a node and then adds the children passed in
-  set: (children) -> this.clear().add(children)
+  set: (children) ->
+    # The use of Promise.resolve can delay when the replacement happens, so
+    # check to only do this when needed. This makes testing nicer when adding
+    # non-promise content
+    if children.then
+      Promise.resolve(children).then((sel) => this.clear().add(sel))
```

Comment:
I was wondering the same, but it's not so easy to do for the `add`/`append` methods since the order in which the element is added would depend on when the promises resolve. 

For `set` its easy to use promises since all the elements are being replaced - so there are no ordering issues. 

For `replace` it is also okay since you know the position of the currently selected element, so at the point when you want to perform the replace, the position to insert the new content can easily be found.